buildSrc/src/main/kotlin/org/cqfn/save/buildutils/kotlin-library.gradle.kts

```diff
@@ -0,0 +1,96 @@
+package org.cqfn.save.buildutils
+
+/**
+ * Precompiled script plugin, that applies common configuration for a KMP project.
+ * It specifies common targets and sets some common compiler flags.
+ * It creates a number of useful source sets and adds common dependencies.
+ * These source sets can be retrieved in a particular build script and configured further as needed.
+ */
+
+import org.gradle.kotlin.dsl.kotlin
+import org.gradle.nativeplatform.platform.internal.DefaultNativePlatform
+import org.jetbrains.kotlin.gradle.targets.jvm.tasks.KotlinJvmTest
+
+plugins {
+    kotlin("multiplatform")
+    kotlin("plugin.serialization")
+}
+
+kotlin {
+    jvm {
+        compilations.all {
+            kotlinOptions {
+                jvmTarget = "11"
+                freeCompilerArgs = freeCompilerArgs + "-Xjvm-default=all"
+            }
+        }
+    }
+    val nativeTargets = listOf(linuxX64(), mingwX64(), macosX64())
```

Comment:
Why diktat didn't warn on this line? the first usage of this variable is at 82 line. Something wrong with diktat configuration?

Reply:
Funny thing is, that here we need to declare this variable here to activate targets before disabling some of them in the next `if`. Speaking about diktat, this behavior is intended: 
 https://github.com/cqfn/diKTat/blob/70066f60425d7a092427e783e278fbfa40bebe83/diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/identifiers/LocalVariablesRule.kt#L204
We are whitelisting several methods, inlcuding `listOf`